src/main/java/com/google/sps/servlets/TaskServlet.java

```diff
@@ -29,9 +37,75 @@
 @WebServlet("/tasks")
 public class TaskServlet extends HttpServlet {
   @Override
+  // doGet method retrieves tasks from datastore and responds with the HTML for each task fetched
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Serve the GET request sent from home page to fetch all the tasks
-    return;
+    UserService userService = UserServiceFactory.getUserService();
+    boolean userLoggedIn = userService.isUserLoggedIn();
+    String userId = userLoggedIn ? userService.getCurrentUser().getUserId() : "null";
+
+    String zipcode = request.getParameter("zipcode");
+    String country = request.getParameter("country");
```

Comment:
Maybe you can add an error checking statement here for the case that zipcode and country is empty or invalid. request.getParameter() can return null if it fails to get the parameter specified and at that time you can send an error.

Reply:
That's smart, I'll implement some error checking in here, thanks!